File: api/Hospital.Api/Dtos/Farmacia/Caja.cs
using System;

namespace Hospital.Api.Dtos.Caja
{
    public class CobroItemDto
    {
        public int IdReferencia { get; set; }   
        public string Origen { get; set; } = string.Empty; 
        public string Paciente { get; set; } = string.Empty;
        public string Concepto { get; set; } = string.Empty; 
        public decimal MontoTotal { get; set; }
        public DateTime Fecha { get; set; }
        public string Estatus { get; set; } = string.Empty;
        
        public bool EsPagado 
        {
            get 
            {
                var s = Estatus.ToLower();
                return s.Contains("pagada") || 
                       s.Contains("atendida") || 
                       s.Contains("entregado") ||
                       s == "pagado";
            }
        }
    }
}

File: web/hospital-web/src/api/caja.ts
import { http } from "./httpClient";

export interface CobroItemDto {
  idReferencia: number;
  origen: "Cita" | "Farmacia";
  paciente: string;
  concepto: string;
  montoTotal: number;
  fecha: string;
  estatus: string;
  esPagado: boolean;
}

export async function getHistorialCaja(desde: string, hasta: string) {
  const params = new URLSearchParams();
  // Enviamos formato YYYY-MM-DD
  params.append("fechaInicio", desde);
  params.append("fechaFin", hasta);

  const res = await http.get<CobroItemDto[]>(`/api/Caja/historial?${params.toString()}`);
  return res.data;
}